### DISCUSSION TITLE
Doesn't appear to compile on cygwin64.

### DISCUSSION THREAD

**Discussion by aotj** [drmortalwombat/oscar64#58] [Q&A]
Here's the result... I did add the windows path to PATH but this is the error I get when I run from the README.  I have gcc version 13+, not sure of make version, but I cloned the repo locally and launched:
from my build directory
make -f ../make/makefile

and the result:

$ make -f ../make/makefile
c++ -c -g -O2 -std=c++11 -Wno-switch ../oscar64/ByteCodeGenerator.cpp -o ByteCodeGenerator.o
In file included from ../oscar64/Array.h:4,
                 from ../oscar64/InterCode.h:3,
                 from ../oscar64/ByteCodeGenerator.h:3,
                 from ../oscar64/ByteCodeGenerator.cpp:1:
../oscar64/MachineTypes.h: In function ‘char* _fullpath(char*, const char*, size_t)’:
../oscar64/MachineTypes.h:64:16: error: ‘realpath’ was not declared in this scope; did you mean ‘relPath’?
   64 |         return realpath(relPath, absPath);
      |                ^~~~~~~~
      |                relPath
make: *** [../make/makefile:25: ByteCodeGenerator.o] Error 1

Any ideas?  Any help would be appreciated.  Just trying to migrate off of c65 for the lack of performance and floats...
Thanks,

**Comment by drmortalwombat** [comment]
realpath() is a posix.1-2008 function, so should be present on all "recent" systems that claim posix compatibility.

This page may help:
https://cygwin.cygwin.narkive.com/xSNPS7yd/realpath-function-not-declared-in-scope-when-std-c-11-option-added-to-g

**Comment by aotj** [comment]
all good .   compiled on linux... I'll worry about cygwin later :)

Respectfully,

Jaman 

    On Sunday, April 14, 2024 at 02:33:46 AM MST, drmortalwombat ***@***.***> wrote:  
 
 


realpath() is a posix.1-2008 function, so should be present on all "recent" systems that claim posix compatibility.

This page may help:
https://cygwin.cygwin.narkive.com/xSNPS7yd/realpath-function-not-declared-in-scope-when-std-c-11-option-added-to-g
